python/lsst/ts/externalscripts/make_LOVE_stress_tests.py

```diff
@@ -0,0 +1,299 @@
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+
+__all__ = ["StressLOVE"]
+
+import asyncio
+import json
+
+import aiohttp
+import requests
+import yaml
+from astropy.time import Time
+from lsst.ts import salobj
+
+
+class ClientInstance:
+    """Connect to a LOVE-manager instance.
+
+    Notes
+    -----
+    **Details**
+
+    * Generate websocket connections using provided credentials
+    by token authentication and subscribes to every
+    event and telemetry specified.
+    """
+
+    def __init__(self, location, username, password, event_streams, telemetry_streams):
+        self.location = location
+        self.username = username
+        self.password = password
+        self.event_streams = event_streams
+        self.telemetry_streams = telemetry_streams
+
+        self.websocket_url = ""
+        self.received_messages = 0
+        self.msg_traces = []
+
+    def request_token(self):
+        """Authenticate on the LOVE-manager instance
+        to get an authorization token."""
+
+        url = f"http://{self.location}/manager/api/get-token/"
+        data = {
+            "username": self.username,
+            "password": self.password,
+        }
+        resp = requests.post(url, data=data)
+        try:
+            token = resp.json()["token"]
+        except Exception:
+            raise Exception("Authentication failed")
+        self.websocket_url = (
+            f"ws://{self.location}/manager/ws/subscription?token={token}"
+        )
+
+    async def handle_message_reception(self):
+        """Handles the reception of messages."""
+
+        if self.websocket:
+            async for message in self.websocket:
+                if message.type == aiohttp.WSMsgType.TEXT:
+                    msg = json.loads(message.data)
+                    if "category" not in msg or (
+                        "option" in msg and msg["option"] == "subscribe"
+                    ):
+                        continue
+                    self.received_messages = self.received_messages + 1
+                    tracing = msg["tracing"]
+                    tracing["client_rcv"] = Time.now().tai.datetime.timestamp()
+                    self.msg_traces.append(tracing)
+
+    async def subscribe_to(self, csc, salindex, stream, topic_type):
```

Comment:
This overloads the term `csc`: `start_ws_client` uses `csc` to mean the CSC name + index and here `csc` means just the name. Please use a different variable name here or in `start_ws_client` to clarify the difference.